runhouse/resources/hardware/cluster.py

```diff
@@ -2330,7 +2330,10 @@ def list(
 
     def list_processes(self):
         """List all workers on the cluster."""
-        return self.client.list_processes()
+        if self.on_this_cluster():
+            return obj_store.get_all_initialized_servlet_names()
+        else:
+            return self.client.list_processes()
```

Comment:
Can we make these APIs match? I prefer for things to match between obj_store, servlet, and http_client because we have so many parallel methods across them to keep track of